cores/arduino/WString.cpp

```diff
@@ -492,7 +492,13 @@ int String::compareTo(const String &s) const
 		if (buffer && len > 0) return *(unsigned char *)buffer;
 		return 0;
 	}
-	return strcmp(buffer, s.buffer);
+	const char *p1 = buffer;
+	const char *p2 = s.buffer;
+
+	while (*p1 == *p2++)
+		if ('\0' == *p1++)
+			return 0;
+	return (*(const char *)p1 - *(const char *)(p2 - 1));
```

Comment:
Wuh.. this is not okay. I see a few more code changes in the PR. Perhaps the PR needs to be rebased to the tip?